src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -89,9 +160,41 @@ public SocksBeginExBuilder typeId(
         }
 
         public SocksBeginExBuilder address(
-            String address)
+            String address) throws UnknownHostException
         {
-            this.beginExRW.address(address);
+            if (IPV4_ADDRESS_MATCHER.get().reset(address).matches())
+            {
+                final Matcher ipv4Matcher = IPV4_ADDRESS_MATCHER.get();
+                final byte[] ipv4AddressBytes = IPV4_ADDRESS_BYTES.get();
+                for (int group = 1; group < ipv4AddressBytes.length + 1; group++)
+                {
+                    ipv4AddressBytes[group - 1] = parseByte(ipv4Matcher.group(group), 10);
+                }
```

Comment:
Hoist lines `169-172` into method called `encodeIpv4AddressBytes` and reuse for `routeEx`.

Reply:
done